app/models/avis.rb

```diff
@@ -50,7 +50,7 @@ class Avis < ApplicationRecord
   scope :updated_since?, -> (date) { where('avis.updated_at > ?', date) }
   scope :discarded_termine_expired, -> { unscope(:joins).where(dossier: Dossier.discarded_termine_expired) }
   scope :discarded_en_construction_expired, -> { unscope(:joins).where(dossier: Dossier.discarded_en_construction_expired) }
-
+  scope :not_hidden_by_administration, -> { where(dossier: Dossier.not_hidden_by_administration) }
```

Comment:
Comme je ne connais pas cette syntaxe, je pose une question peut être naïve, mais ça fait bien un join et pas une sub query de tous les dossiers visibles ? Tu peux faire un `Avis.not_hidden_by_administration.to_sql` et le poster ici ?

Reply:
`SELECT \"avis\".* FROM \"avis\" INNER JOIN \"dossiers\" ON \"dossiers\".\"hidden_at\" IS NULL AND \"dossiers\".\"id\" = \"avis\".\"dossier_id\" WHERE \"avis\".\"dossier_id\" IN (SELECT \"dossiers\".\"id\" FROM \"dossiers\" WHERE \"dossiers\".\"hidden_at\" IS NULL AND \"dossiers\".\"hidden_by_administration_at\" IS NULL)`

@tchak

Reply:
Euh, donc ça fait une sub query et pas un join. Moi ça me fait peur. La sub query ce sont des millions de dossiers... Vous avez un avis @mfo et @LeSim ?

Reply:
Je ne sais pas comment le moteur postgres va optimiser cette requete, faudrait essayer un avec `explain analyze` et comparer les perfs.

Mais j'avoue que naivement, j'essayerai de faire un seul join avec 2 conditions `hidden_at IS NULL AND hidden_by_administration_at IS NULL`

ca doit s'ecrire

```suggestion
  scope :not_hidden_by_administration, -> { where(dossiers: { hidden_by_administration_at: nil }) }
```

Reply:
J'ai fait la modif @LeSim @tchak 

`SELECT \"avis\".* FROM \"avis\" INNER JOIN \"dossiers\" ON \"dossiers\".\"hidden_at\" IS NULL AND \"dossiers\".\"id\" = \"avis\".\"dossier_id\" WHERE \"dossiers\".\"hidden_by_administration_at\" IS NULL`

Reply:
> Euh, donc ça fait une sub query et pas un join. Moi ça me fait peur. La sub query ce sont des millions de dossiers... Vous avez un avis @mfo et @LeSim ?

j'ai regardé le explain (a moins du verbose c'etait flou), mais ca rajoute une étape d'index scan don on se passerait bien ; 

>         ->  Index Scan using dossiers_pkey on public.dossiers  (cost=0.43..2.67 rows=1 width=4) (actual >time=0.012..0.012 rows=1 loops=377946)
>               Output: dossiers.*
>               Index Cond: (dossiers.id = avis.dossier_id)
>               Filter: (dossiers.hidden_at IS NULL)
>               Worker 0:  actual time=0.012..0.012 rows=1 loops=125195
>               Worker 1:  actual time=0.012..0.012 rows=1 loops=124810